### ISSUE TITLE
Too long Output Makes the Job Fail

### ISSUE THREAD

**Issue by srkunze** [jenkinsci/ansicolor-plugin#61]
We run a salt-ssh command which we pipe through tee. This preserves the color codes but somehow the job crashes. When the plugin is not active for the job the job runs well.

We cut the output in order to use the plugin (we like colors ;) ), but it seems like a bug.

**Comment by dblock** [comment]
You should be able to narrow down to the line of output that causes this. Compare the output with and without the plugin?

**Comment by srkunze** [comment]
Thanks for your reply.

Unfortunately, the line of output that causes the job to fail **varies**.

We started the very same job with the very same parameters (and we know it should produce the very same output), and the job fails at different lines.

Our first guess was that the sheer number of lines could be an issue. After we truncated the output (silenced some shell cmds), everything worked just fine. This behavior is reproducible.

This example is the successful (not colorized) output:

                      ...
                      [0;32mLinking '/home/..../static/tiny_mce/plugins/margin/editor_plugin_src.js'[0;0m
                      [0;32mLinking '/home/..../static/tiny_mce/plugins/margin/langs/en.js'[0;0m
                      [0;32mLinking '/home/....tinymce/static/tiny_mce/plugins/margin/langs/en_dlg.js'[0;0m
                      [0;32mLinking '/home/..../tinymce/static/tiny_mce/plugins/margin/langs/de.js'[0;0m
                      [0;32mLinking '/home/..../tinymce/static/tiny_mce/plugins/margin/js/dialog.js'[0;0m
                      [0;32mLinking '/home/..../tinymce/static/tiny_mce/plugins/margin/img/top-add.gif'[0;0m
                      [0;32mLinking '/home/..../tinymce/static/tiny_mce/plugins/margin/img/top-sub.gif'[0;0m
                      [0;32mLinking '/home/..../tinymce/static/tiny_mce/plugins/tinymce/css/tinymce.css'[0;0m
                      [0;32m[0;0m
                      [0;32m2164 static files symlinked to '/home/..../static'.[0;0m[0;0m


This is how the (colorized) failure looks like. Note, the 'tee' is part of our command.

                      Linking '/home/..../django_jquery/static/django_jquery/jquery-ui.lightness/images/ui-icons_222222_256x240.png'
                      Linking '/home/..../selectable/static/selectable/js/jquery.dj.selectable.js'
                      Linking '/home/..../selectable/static/selectable/css/dj.selectable.css'
                      Linking '/home/..../modwotee: write error
    Build step 'Execute shell' marked build as failure
    Description set: 
    Finished: FAILURE


Note: The number of spaces before each line is authentic. The output is a series of thousands of lines alike the ones present here. As far as I can tell they are completely similar regarding structure and color codes.

**Comment by dblock** [comment]
I think the next step is to try to reproduce this in a unit test in this project. If this is a size problem we should be able to pipe a very large string through the plugin and see what happens.

**Comment by srkunze** [comment]
@dblock So, given these examples, there is nothing suspicious, right?

**Comment by dblock** [comment]
Right.

**Comment by srkunze** [comment]
Alright. Well, how do we start here? Currently, I have no idea how to write a proper unit test for this project.

**Comment by dblock** [comment]
Get the project to build with maven / Eclipse to start. You might want to edit https://github.com/dblock/jenkins-ansicolor-plugin/blob/master/CONTRIBUTING.md as you go for the next person. I'll help.

**Comment by qvicksilver** [comment]
This is a good resource on how to write unit tests for Jenkins stuff: https://wiki.jenkins-ci.org/display/JENKINS/Unit+Test.
I attempted to feed a rather long string (14m chars) from /dev/urandom through the plugin by modifying the testWorkflowWrap test from AnsiColorBuildWrapperTest.java but that did not result in any faliures.

**Comment by srkunze** [comment]
@dblock @qvicksilver Thanks so far for your support. :)

I tried a different approach to get a better idea of what's going on. I changed the jenkins shell script from

    my_cmd | tee out.log

to

    my_cmd > out.log
    cat out.log

Surprisingly, the latter variant works with the plugin. So, we **could** close that issue.

However, I get the feeling that this cannot be right. Both `tee` and `cat` are basic Unix commands. How do they differ in such a way that it crashes the plugin?
